spec/rubocop/cop/rails/http_positional_arguments_spec.rb

```diff
@@ -0,0 +1,227 @@
+        expect(cop.offenses.size).to eq(0)
+      end
+    end
+  end
+
+  describe :patch do
+    let(:source) do
+      <<-EOS
+patch :update,
+          id: @user.id,
+          ac: {
+            article_id: @article1.id,
+            profile_id: @profile1.id,
+            content: 'Some Text'
+          }
+EOS
+    end
+
+    let(:corrected_result) do
+      <<-EOS
+patch :update, params: { id: @user.id, ac: {
+            article_id: @article1.id,
+            profile_id: @profile1.id,
+            content: 'Some Text'
+          } }
+EOS
+    end
+
+    it 'records an offense' do
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(1)
+    end
+
+    it 'autocorrects offense' do
+      new_source = autocorrect_source(cop, source)
+      expect(new_source).to eq(corrected_result)
+    end
+  end
+
+  describe :post do
+    let(:source) do
+      <<-EOS
+post :create,
+          id: @user.id,
+          ac: {
+            article_id: @article1.id,
+            profile_id: @profile1.id,
+            content: 'Some Text'
+          }
+      EOS
+    end
+
+    let(:corrected_result) do
+      <<-EOS
+post :create, params: { id: @user.id, ac: {
+            article_id: @article1.id,
+            profile_id: @profile1.id,
+            content: 'Some Text'
+          } }
+      EOS
+    end
+
+    it 'records an offense' do
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(1)
+    end
+
+    it 'autocorrects offense' do
+      new_source = autocorrect_source(cop, source)
+      expect(new_source).to eq(corrected_result)
+    end
+  end
+
+  %w(post get patch put delete).each do |keyword|
+    it 'does not record an offense when keyword found on variable' do
+      source = "@user.#{keyword}.id = ''"
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(0)
+    end
+
+    it 'does not record an offense when keyword is found on object' do
+      source = "obj.#{keyword}.profile.id"
+      inspect_source(cop, source)
+      expect(cop.offenses.size).to eq(0)
+    end
+  end
+
+  it 'auto corrects controller method when it is a method' do
+    source = 'post user_attrs, id: 1'
+    inspect_source(cop, source)
+    expect(cop.offenses.size).to eq(1)
+    new_source = autocorrect_source(cop, source)
+    expected = 'post user_attrs, params: { id: 1 }'
+    expect(new_source).to eq(expected)
+  end
+
+  it 'auto corrects controller method when it is a symbol' do
+    source = 'post :user_attrs, id: 1'
+    inspect_source(cop, source)
+    expect(cop.offenses.size).to eq(1)
+    new_source = autocorrect_source(cop, source)
+    expected = 'post :user_attrs, params: { id: 1 }'
+    expect(new_source).to eq(expected)
+  end
+
+  it 'should not return offense when post is found' do
+    source = "if post.stint_title.present? \n true\n end"
+    inspect_source(cop, source)
+    expect(cop.offenses.size).to eq(0)
+  end
+
+  it 'should not remove quotes when single quoted' do
```

Comment:
does not auto-correct

Such tests are redundant if you've already tested this doesn't produce offenses.